### ISSUE TITLE
Implement a better CSV reading functionality

### ISSUE THREAD

**Issue by inejc** [picnicml/doddle-model#38]
Currently, data can be read from CSV files with [this piece of code](https://github.com/picnicml/doddle-model/blob/master/src/main/scala/io/picnicml/doddlemodel/data/package.scala#L35), but the implementation is very limited; only numerical data (doubles) can be read, non-numerical fields result in `Double.NaN` which is used to encode missing data in doddle-model. The new implementation should be able to encode numerical and categorical variables with missing values, i.e. numerical features should be encoded as doubles directly and categorical features should first be encoded to numerical representation (take a look at [label ecoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html)) and only then converted to doubles. The function will probably look something like `loadCsvDataset(filePath: String, naString: String, headerLine: Boolean = true): DenseMatrix[Double]`.

A 3rd party library should be used to parse CSV files (take a look at [this discussion](https://www.reddit.com/r/scala/comments/5tvnhy/what_is_the_best_csv_parser_for_scala/) for some starting points).
